qadence/model.py

```diff
@@ -432,6 +432,56 @@ def _from_dict(cls, d: dict, as_torch: bool = False) -> QuantumModel:
 
         return qm
 
+    def load_params_from_dict(self, d: dict, strict: bool = True) -> None:
+        """Copy parameters from dictionary into this QuantumModel.
+
+        Unlike :meth:`~qadence.QuantumModel.from_dict`, this method does not create a new
+        QuantumModel instance, but rather loads the parameters into the same QuantumModel.
+        The behaviour of this method is similar to :meth:`~torch.nn.Module.load_state_dict`.
+
+        The dictionary is assumed to have the format as saved via
+        :meth:`~qadence.QuantumModel.to_dict`
+
+        Args:
+            d (dict): The dictionary
+            strict (bool, optional):
+                Whether to strictly enforce that the parameter keys in the dictionary and
+                in the model match exactly. Default: ``True``.
+        """
+        param_dict = d["param_dict"]
+        missing_keys = []
+        unexpected_keys = []
```

Comment:
You can use here `unexpected_keys = set(self._params.keys()) - set(param_dict.keys())` and I would then check if strict and directly raise the error before looping. Loop can then only be done over the present keys. Just a small trick to avoid unnecessary instructions.

Reply:
Wow that's really good! Thanks for that, I'll add it now for both cases and throw the error before